ssl/statem/statem_clnt.c

```diff
@@ -1128,6 +1129,7 @@ MSG_PROCESS_RETURN tls_process_server_hello(SSL *s, PACKET *pkt)
             if (!ssl_get_new_session(s, 0)) {
                 goto f_err;
             }
+            s->session->ssl_version = s->version;
```

Comment:
> That means s->session is initialized from line 831. But that's before version negotiation!

This could also mean EAP-FAST case, and I don't want to allow stale session there. (stale = diff. protocol version)

> I feel like I haven't been explaining the issue very well, so let me know if it isn't. The code is pretty gnarly. :-)

Yeah, I guess I figured it out. Sorry for my stupidness! Added an additional check for EAP-FAST case, though.